src/core/qgstracer.cpp

```diff
@@ -585,6 +629,12 @@ void QgsTracer::setDestinationCrs( const QgsCoordinateReferenceSystem &crs, cons
   invalidateGraph();
 }
 
+void QgsTracer::setRenderContext( const QgsRenderContext *renderContext )
+{
+  mRenderContext = std::unique_ptr<QgsRenderContext>( new QgsRenderContext( *renderContext ) );
+  invalidateGraph();
```

Comment:
If I read the code correctly, this introduces a performance problem - every single time we need the graph it will regenerated due to this invalidation... (when asked for shortest path (that's on every mouse move) we call init() which calls configure() which _may_ invalidate the graph if something important has changed)

Please test with a larger layer where graph creation takes a bit of time to make sure that tracing is still fast...

Reply:
> Please test with a larger layer where graph creation takes a bit of time to make sure that tracing is still fast...

For my tests (with a large dataset befor I got the warning on max limit data) I didn't know how to see any real difference.